python/hsfs/core/training_dataset_job_conf.py

```diff
@@ -71,5 +72,5 @@ def to_dict(self):
             ]
             if self._write_options
             else None,
-            "sparkJobConfiguration": self._spark_job_configuration,
+            JobConfiguration.DTO_TYPE: JobConfiguration(**self._spark_job_configuration if self._spark_job_configuration else {}).to_dict(),
```

Comment:
I assume this is needed because we explicitly need to call the `to_dict()` method to inject the default?

Wouldn't it be easier to do the injection in the init and leave this code as it is?